### ISSUE TITLE
Not receiving sent TCP packets 

### ISSUE THREAD

**Issue by zachary822** [stephenwvickers/node-raw-socket#7]
Below is some code that I wrote. For the life of me, I can't get anything from the `on.("message", func)`. Am I doing something wrong? I have confirmed that I have received a packet via wireshark, but it's not showing up here.
```
var raw = require('raw-socket');
var crypto = require('crypto');
var ip = require('ip');

var s = raw.createSocket({
    protocol: raw.Protocol.TCP,
    generateChecksums: true
});

// source port 20 to destination port 80. Packet length 20 with SYN flag set. Other flags are zeroed.
var p = new Buffer('0014005000000000000000005002200000000000', 'hex');
// generate four random bytes for the packet sequence.
crypto.randomBytes(4).copy(p, 4);

//build pseudo TCP header.
var genPseudoHeader = function (srcIp, dstIp, tcpPacketLength) {
    var pseudoHeader = new Buffer(12);
    pseudoHeader.fill(0);
    pseudoHeader.writeUIntBE(ip.toLong(srcIp), 0, 4);
    pseudoHeader.writeUIntBE(ip.toLong(dstIp), 4, 4);
    pseudoHeader.writeUIntBE(6, 9, 1);
    pseudoHeader.writeUIntBE(tcpPacketLength, 10, 2);
    return pseudoHeader;
};

// this gets absolutely nothing.
s.on('message', function (buffer, address) {
    console.log("received " + buffer.length + " bytes from " + address + ": " + buffer.toString("hex"));
});

// calculate checksum using utility function.
var sum = raw.createChecksum(genPseudoHeader('192.168.40.100', '192.168.40.1', p.length), p);
// writing checksum to TCP packet.
p.writeUIntBE(sum, 16, 2);
console.log(p);

function afterSend(error, bytes) {
    if (error) {
        console.log (error.toString ());
    }
    else {
        console.log ("sent " + bytes + " bytes");
    }
}

// send packet
s.send(p, 0, p.length, "192.168.40.1", function () {}, afterSend);
```

**Comment by zhaowenjian** [comment]
can not install ?
npm ERR! Linux 4.2.0-16-generic
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "raw-socket"
npm ERR! node v6.0.0-pre
npm ERR! npm  v3.3.6
npm ERR! path /home/foreve_/workspace/ipTrafficMonitor/node_modules/node-raw-socket
npm ERR! code EISGIT

npm ERR! git Appears to be a git repo or submodule.
npm ERR! git     /home/foreve_/workspace/ipTrafficMonitor/node_modules/node-raw-socket
npm ERR! git Refusing to remove it. Update manually,
npm ERR! git or move it out of the way first.

npm ERR! Please include the following file with any support request:
npm ERR!     /home/foreve_/workspace/ipTrafficMonitor/npm-debug.log

**Comment by stephenwvickers** [comment]
Hi,

Unfortunately this is not enough information to assist.

Can you please provide a Wireshark dump showing the request you send, and the response the application does not see?

I believe your installation issue is covered under:

https://github.com/stephenwvickers/node-raw-socket/issues/8

Steve

**Comment by stephenwvickers** [comment]
Resolved, as per issue #8

**Comment by zachary822** [comment]
This is the output:
```
<Buffer 00 14 00 50 71 b4 b3 c7 00 00 00 00 50 02 20 00 4c 96 00 00>
sent 20 bytes
```
Which just prints the sent bytes from the `afterSend()` function.

Capture file attached. A syn/ack is shown to be received. (open with wireshark)
[syn-ack.txt](https://github.com/stephenwvickers/node-raw-socket/files/36463/syn-ack.txt)
